components/org.wso2.carbon.identity.conditional.auth.functions.opa/src/test/java/org/wso2/carbon/identity/conditional/auth/functions/opa/InvokeOpaFunctionImplTest.java

```diff
@@ -0,0 +1,203 @@
+        import org.powermock.core.classloader.annotations.PowerMockIgnore;
+        import org.powermock.core.classloader.annotations.PrepareForTest;
+        import org.powermock.modules.junit4.PowerMockRunner;
+        import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+        import org.testng.IObjectFactory;
+        import org.testng.annotations.AfterMethod;
+        import org.testng.annotations.BeforeMethod;
+        import org.testng.annotations.ObjectFactory;
+        import org.testng.annotations.Test;
+        import org.wso2.carbon.identity.application.authentication.framework.AsyncProcess;
+        import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.JsGraphBuilder;
+        import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticatedUser;
+        import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticationContext;
+        import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsStep;
+        import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsSteps;
+        import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+        import org.wso2.carbon.identity.conditional.auth.functions.common.utils.ConfigProvider;
+        import org.wso2.carbon.identity.conditional.auth.functions.opa.InvokeOpaFunctionImpl;
+
+        import java.io.ByteArrayInputStream;
+        import java.util.HashMap;
+        import java.util.Map;
+
+        import static org.mockito.Matchers.anyString;
+        import static org.mockito.MockitoAnnotations.initMocks;
+        import static org.powermock.api.mockito.PowerMockito.*;
+
+        @RunWith(PowerMockRunner.class)
+        @PrepareForTest({JsGraphBuilder.class, HttpClientBuilder.class, RequestConfig.class, RequestConfig.Builder.class, AsyncProcess.class})
+        @PowerMockIgnore("jdk.internal.reflect.*")
+        public class InvokeOpaFunctionImplTest {
+
+            @Mock
+            private HttpClientBuilder mockBuil, mockBuilCon;
+
+            @Mock
+            private RequestConfig mockCon;
+
+            @Mock
+            private ConfigProvider conPro;
+
+            @Mock
+            private RequestConfig.Builder bui, bui1, conTime, reqTime, sockTime;
+
+
+            @Mock
+            private AsyncProcess mockAPro;
+
+
+            private CloseableHttpClient httpClient;
+            private HttpPost request;
+            private HttpEntity entity;
+            private StatusLine statusline;
+            private CloseableHttpResponse response;
+            private InvokeOpaFunctionImpl serviceInvoker;
+
+
+//            @BeforeMethod(alwaysRun = true)
+            @BeforeMethod
+            public void setup() throws Exception {
+                initMocks(this);
+                this.httpClient = mock(CloseableHttpClient.class);
```

Comment:
if we use @Mock annotation for the variables that need to be mocked, initMocks() will create mock instances for all. Then no need to manually mock them like this
mock(CloseableHttpClient.class);